frontend/packages/core/src/stepper.tsx

```diff
@@ -227,22 +226,23 @@ const Stepper = ({
           variant: "pending" as StepIconVariant,
           nonLinear,
         };
+        const { isCompleted } = step.props;
 
-        if (completed) {
-          if (completed[idx]) {
-            stepProps.variant = "success";
-          } else if (idx === activeStep) {
-            stepProps.variant = step.props.error ? "failed" : "active";
-          }
+        if (isCompleted) {
```

Comment:
Looks like we can combine this a bit like:

```js
        if (idx === activeStep) {
          stepProps.variant = step.props.error ? "failed" : "active";
        } else if (isComplete || (!nonLinear && idx < activeStep)) {
          stepProps.variant = "success";
        }
```

Reply:
I'm thinking in the case when the user returns to a previous completed step, with your proposal if you select a completed step, the checkmark is replaced with the "active" state.

Is this how we want it to work?

Reply:
We could just invert the conditions to avoid the previous behavior:

```
if (isComplete || (!nonLinear && idx < activeStep)) {
  stepProps.variant = "success";
} else if (idx === activeStep) {
  stepProps.variant = step.props.error ? "failed" : "active";
}
```